src/Mafia/Process.hs

```diff
@@ -174,6 +182,36 @@ renderProcessError = \case
 
 ------------------------------------------------------------------------
 
+createProcess :: MonadIO m => Process.CreateProcess -> m (Maybe Handle, Maybe Handle, Maybe Handle, Process.ProcessHandle)
+createProcess = liftIO . Process.createProcess
+
+-- Spawn a new process, and if we get a ctrl-c, make absolutely sure everything we started is finished.
+createProcessAnnihilate :: MonadIO m => Process.CreateProcess -> m (Maybe Handle, Maybe Handle, Maybe Handle, Process.ProcessHandle)
+createProcessAnnihilate cp = do
+  (a, b, c, pid) <- createProcess cp { Process.create_group = True }
+
+  pgid <- liftIO $ ProcessInternals.withProcessHandle pid getPgid
+  case pgid of
+   Just pgid' -> do
+    _ <- liftIO $ Signals.installHandler Signals.keyboardSignal (Signals.Catch $ killer pgid') Nothing
```

Comment:
Sidenote, `keyboardSignal` has to be the worst alias for `SIGINT` I've ever seen.